analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/DeliveringDebugFeaturesInProduction.cs

```diff
@@ -42,5 +42,20 @@ public sealed class DeliveringDebugFeaturesInProduction : DeliveringDebugFeature
         private bool IsDevelopmentCheck(SyntaxNode node, SemanticModel semanticModel) =>
             node is InvocationExpressionSyntax condition
             && IsValidationMethod(semanticModel, condition, condition.Expression.GetIdentifier()?.ValueText);
+
+        private static SyntaxNode EnclosingScope(SyntaxNode node) =>
```

Comment:
Move the method to SonarAnalyzer.CSharp -> SyntaxNodeExtensions (and ideally bring it under test for all kinds of code blocks).